service/frontend/workflowHandler.go

```diff
@@ -3087,9 +3043,7 @@ func (wh *WorkflowHandler) getLoggerForTask(taskToken []byte) log.Logger {
 
 // startRequestProfile initiates recording of request metrics
 func (wh *WorkflowHandler) startRequestProfile(scope int) (metrics.Scope, metrics.Stopwatch) {
-	wh.startWG.Wait()
-
-	metricsScope := wh.metricsClient.Scope(scope).Tagged(metrics.DomainUnknownTag())
+	metricsScope := wh.GetMetricsClient().Scope(scope).Tagged(metrics.DomainUnknownTag())
```

Comment:
could you plz test the perf diff between `wh.metricsClient` vs `wh.GetMetricsClient()`?

Reply:
Each getter method takes 1.7ns. Compared to other operation it's basically nothing. My concerns with creating a separate field for `metricsClient` is that people may be confused and not sure why there's two ways to get the metricsClient.